### ISSUE TITLE
Make Soledad usage of CouchDB less aggressive

### ISSUE THREAD

**Issue by shyba** [pixelated/pixelated-platform#105]
This issue is about giving breath do CouchDB, as Soledad today is overloading it on some requests. Changes were already planned for getting this better at LEAP side, I am creating this issue here to keep track with what we can help.
The changes will impact sync_state and sync_log on Soledad server side. This issue tracks sync_state for now, as we are helping on it.
This also impacts Wazokazi, since on last attemp with v0.6 CouchDB load was too high. This issue is one of the causes.

Steps
- [x] Spike a sync_state logic in memory, outside of CouchDB (draft)
- [x] Migrate that to Beaker and add it as a dependency
- [ ] Refactor and clean up

**Comment by shyba** [comment]
PR Created: https://github.com/leapcode/soledad/pull/264

**Comment by shyba** [comment]
sync_state -> solved on beaker
sync_log -> solved by drebs, on beaker
caching -> bonus
I've picked up his commit into this branch, applied some more caching and tested on unstable looking into CouchDB log.
Good to go. Refactor isn't possible on current branch (would break leap_mx). Will do on another PR using another card as well.

**Comment by fbernitt** [comment]
Some questions from looking at the diffs:

- Why defining cache as a property and using it from within the class instead of self._cache? 
- In ../server/../auth.py - Shouldn't we also validate the token, not only the uuid? Not sure, but wouldn't this currently mean that I could hijack an ongoing sync process by just knowing the uuid? (also commented on the PR)
- It seems the beaker cache is currently always initialised with memory? How could I change that using a config file? Or is that something for the future?

**Comment by shyba** [comment]
* That is a side effect of couch.py holding CouchDatabase inside common while caching is on server. I started refactoring it, but realized that it would break leap_mx and can become a change bigger than that. Will do on next PR, as explained on comments. :smile: 
* reverted, good catch. This needs caching, but totally lacks tests. Auth without tests is too dangerous to modify. Will add tests and cache on another isolated change (easier to review, since this is risky)
* yes, using a file or config implies in checking permissions, which can imply in a broader discussion outside of this scope. Before starting a discussion about that I decided to implement the simplest step possible.
Thank you for helping on review!
